src/main/kotlin/app/revanced/patches/tiktok/interaction/downloads/patch/DownloadsPatch.kt

```diff
@@ -50,7 +60,78 @@ class DownloadsPatch : BytecodePatch(
                 return v0
             """
         )
+        //Change the download path patch
+        val method4 = DownloadPathParentFingerprint.result!!.mutableMethod
+        val implementation4 = method4.implementation
+        val instructions = implementation4!!.instructions
+        var targetOffset = -1
+        //Search for the target method called instruction offset.
+        run found@{
+            instructions.forEachIndexed { index, instruction ->
+                if (instruction.opcode != Opcode.CONST_STRING) return@forEachIndexed
+                val reference = (instruction as ReferenceInstruction).reference as StringReference
+                if (reference.string != "video/mp4") return@forEachIndexed
+                val targetInstruction = instructions[index + 1] as ReferenceInstruction
+                if (targetInstruction.opcode != Opcode.INVOKE_STATIC) return@forEachIndexed
+                targetOffset = index + 1
+                return@found
```

Comment:
```suggestion
                return@forEachIndexed
```

Reply:
I use this to break the forEachIndexed loop when found the the targetOffset. Afaik, return@forEachIndexed is similar to `continue` in a for loop.

Reply:
Oh, then your code is correct. But isn't it better to just replace it with a for-loop then?

Reply:
Yeah. I will change it to a for loop.